File: sdks/python/examples/owner_sequence_clipboard_aegis_ctx.py
import os

from cua_sdk import Cua


cua = Cua.connect(profile=os.environ.get("CUA_PROFILE", "default"))
owner = cua.acquire_owner("python owner sequence", ttl_ms=30000)

try:
    cua.create_profile(
        "python-sdk-example",
        mode="supervised",
        duration_ms=60000,
        capabilities={"clipboard": True},
        session=owner,
    )
    cua.activate_profile(session=owner)
    cua.dispatch(
        {
            "schema_version": "cua.v1",
            "kind": "sequence",
            "actions": [
                {"kind": "open_app", "app_name": "Notes"},
                {"kind": "key_press", "combo": "cmd+n"},
            ],
            "inter_action_delay_ms": 120,
        },
        session=owner,
    )
    cua.clipboard_write("hello from cua", session=owner)
    print(cua.clipboard_read(True))
    print(cua.aegis(["--help"], session=owner))
    print(cua.ctx(["--help"], session=owner))
finally:
    cua.cancel_session(owner)

File: sdks/typescript/examples/owner-sequence-clipboard-aegis-ctx.mjs
import { Cua } from "../dist/index.js";

const cua = await Cua.connect({ profile: process.env.CUA_PROFILE ?? "default" });
const owner = await cua.acquireOwner("typescript owner sequence", 30000);

try {
  await cua.createProfile(
    {
      name: "typescript-sdk-example",
      mode: "supervised",
      durationMs: 60000,
      capabilities: { clipboard: true },
    },
    owner,
  );
  await cua.activateProfile(owner);
  await cua.dispatch(
    {
      schema_version: "cua.v1",
      kind: "sequence",
      actions: [
        { kind: "open_app", app_name: "Notes" },
        { kind: "key_press", combo: "cmd+n" },
      ],
      inter_action_delay_ms: 120,
    },
    { session: owner },
  );
  await cua.clipboardWrite("hello from cua", owner);
  console.log(await cua.clipboardRead(true));
  console.log(await cua.aegis(["--help"], { session: owner }));
  console.log(await cua.ctx(["--help"], { session: owner }));
} finally {
  await cua.cancelSession(owner);
}
